src/csharp/Microsoft.Spark/ML/Param/ParamMap.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+
+namespace Microsoft.Spark.ML.Feature.Param
+{
+    /// <summary>
+    /// A param to value map.
+    /// </summary>
+    public class ParamMap : IJvmObjectReferenceProvider
+    {
+        private static readonly string s_ParamMapClassName = "org.apache.spark.ml.param.ParamMap";
+
+        /// <summary>
+        /// Creates a new instance of a <see cref="ParamMap"/>
+        /// </summary>
+        public ParamMap() : this(SparkEnvironment.JvmBridge.CallConstructor(s_ParamMapClassName))
+        {
+        }
+
+        public ParamMap(JvmObjectReference jvmObject) => Reference = jvmObject;
```

Comment:
Should this be public or internal ?  Do you expect users to create ParamMap objects by passing in JvmObjectReference's ?

Reply:
I'll make it internal. Thanks